resources/views/hardware/edit.blade.php

```diff
@@ -491,6 +490,124 @@ function show_er(selector) {
     //console.warn("The Model is: "+model+" and the select is: "+select);
   });
 
+   $("form").submit( function(event) {
+    event.preventDefault();
+    return sendForm();
+  });
+
+  // Resize Files when chosen
+
+
+
+    //First check to see if there is a file before doing anything else
+
+    var imageData = "";
+    var $fileInput = $('#file-upload'); 
+    $fileInput.on('change', function(e) {
+      if( $fileInput != '' ) {
+        if(window.File && window.FileReader && window.FormData) {
+          var file = e.target.files[0];
+          if(file) {
+            if(/^image\//i.test(file.type)) {
+              readFile(file);
+            } else {
+              alert('Invalid Image File :(');
+            }
+          }
+        }
+        else {
+          console.log("File API not supported, not resizing");
+        } 
+      }
+    });
+
+
+  function readFile(file) {
+    var reader = new FileReader();
+
+    reader.onloadend = function() {
+      processFile(reader.result, file.type);
+    }
+
+    reader.onerror = function() { 
+      alert("Unable to read file");
+    }
+
+    reader.readAsDataURL(file);
+  }
+
+  function processFile(dataURL, fileType) {
+    var maxWidth = 800;
+    var maxHeight = 800;
+
+    var image = new Image();
+    image.src = dataURL;
+
+    image.onload = function() {
+      var width = image.width;
+      var height = image.height;
+      var shouldResize = (width > maxWidth) || (height > maxHeight);
+
+      if(!shouldResize) {
+        imageData = dataURL;
+        return;
+      }
+
+      var newWidth;
+      var newHeight;
+
+      if( width > height) {
+        newHeight = height * (maxWidth/width);
+        newWidth = maxWidth;
+      } else {
+        newWidth = width * (maxHeight/height);
+        newHeight = maxHeight;
+      }
+      var canvas = document.createElement('canvas');
+
+      canvas.width = newWidth;
+      canvas.height = newHeight;
+
+      var context = canvas.getContext('2d');
+
+      context.drawImage(this, 0, 0, newWidth, newHeight);
+
+      dataURL = canvas.toDataURL( fileType );
+
+      imageData = dataURL;
+
+    };
+
+    image.onerror = function () {
+      alert('Unable to process file :(');
+    }
+  }
+
+  function sendForm() {
+    var form = $("#create-form").get(0);
```

Comment:
This is just modifying the form thats generated via Laravel Forms higher on the page.  The CSRF token comes from there.  At least if I'm understanding you correctly.